pykoopman/regression/_base_ensemble.py

```diff
@@ -0,0 +1,22 @@
+from sklearn.compose import TransformedTargetRegressor
+
+
+class EnsembleBaseRegressor(TransformedTargetRegressor):
```

Comment:
It looks like you are just renaming TransformedTargetRegressor with EnsembleBaseRegressor, except a few attributes checks.

I looked TransformedTargetRegressor, it seems to merely enabling "nonlinear features on target". Then where is the functionality for "ensemble, non-consecutive data"? 

PS: I saw in the examples that you provided, you are using DMDc regressor, which does not call any regressor from PyDMD at all. The root cause of difficulty in working with non-consecutive data comes from the fact that PyDMD regressor.fit only recognizes consecutive data. 

Do you have any idea on how to solve this?

Reply:
I solved this by using the TransformedTargetRegressor. The scikit-learn BaseEstimator (used for regressors) has x,y as inputs, but the TransformerMixing (used for observables) applies the transformation only to x, not to the target y. This is an intrinsic problem with sci-kit learn itself. They introduced the TransformedTargetRegressor as a way to also transform the target y, which is needed if both data-matrices x,y are provided, which is the case when providing non-consecutive or ensemble data.